plugins/modules/nd_compliance_requirement_config_snapshot.py

```diff
@@ -0,0 +1,201 @@
+
+DOCUMENTATION = r"""
+---
+module: nd_compliance_requirement_config_snapshot
+version_added: "0.2.1"
+short_description: Manage snapshot configuration type compliance requirements
+description:
+- Manage snapshot configuration type on Cisco Nexus Dashboard Insights (NDI).
+author:
+- Akini Ross (@akinross)
+options:
+  insights_group:
+    description:
+    - The name of the insights group.
+    type: str
+    required: yes
+    aliases: [ fab_name, ig_name ]
+  name:
+    description:
+    - The name of the compliance requirement.
+    type: str
+  description:
+    description:
+    - The description of the compliance requirement.
+    type: str
+    aliases: [ descr ]
+  enabled:
+    description:
+    - Enable the compliance requirement.
+    type: bool
+  sites:
+    description:
+    - Names of the Assurance Entities.
+    type: list
+    elements: str
+  baseline_site:
+    description:
+    - Name of the Assurance Entity to set as baseline.
+    type: str
+    aliases: [ site, site_name ]
+  epoch_id:
+    description:
+    - The id of the epoch.
+    - When epoch id is not provided it will retrieve the latest known epoch id.
+    type: str
+  allow_new_configuration_objects:
+    description:
+    - Allow addition of new configuration objects.
+    type: bool
+    default: false
+  state:
+    description:
+    - Use C(query) for retrieving the version object.
+    type: str
+    choices: [ query, absent, present ]
+    default: query
+extends_documentation_fragment: cisco.nd.modules
+"""
+
+EXAMPLES = r"""
+- name: Get all compliance snapshot configuration type requirements
+  cisco.nd.nd_compliance_requirement_config_snapshot:
+    insights_group: igName
+    state: query
+  register: query_results
+- name: Get a specific compliance snapshot configuration type requirement
+  cisco.nd.nd_compliance_requirement_config_snapshot:
+    insights_group: igName
+    name: complianceRequirementName
+    state: query
+  register: query_results
+- name: Create compliance snapshot configuration type requirement
+  cisco.nd.nd_compliance_requirement_config_snapshot:
+    insights_group: igName
+    name: complianceRequirementName
+    sites:
+    - siteName1
+    - siteName2
+    enabled: false
+    epoch_id: 0e5604f9-373a123c-b535-33fc-8d11-672d08f65fd1
+    state: present
+- name: Delete compliance snapshot configuration type requirement
+  cisco.nd.nd_compliance_requirement_config_snapshot:
+    insights_group: igName
+    name: complianceRequirementName
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec, sanitize_dict
+from ansible_collections.cisco.nd.plugins.module_utils.ndi import NDI
+from ansible_collections.cisco.nd.plugins.module_utils.ndi_argument_specs import compliance_base_spec
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(compliance_base_spec())
```

Comment:
Should we use a documentation fragment for this compliance_base_spec part?

Reply:
yes good idea will do